response/writer.go

```diff
@@ -53,9 +53,15 @@ func writeUnmodifiedResponse(ctx context.Context, w http.ResponseWriter, babbage
 }
 
 func writeResponseWithMaxAge(ctx context.Context, w http.ResponseWriter, babbageResponse *http.Response, maxAge int) {
-	cacheControlValue := fmt.Sprintf("max-age=%d", maxAge)
-	additionalHeaders := map[string]string{
-		"Cache-Control": cacheControlValue,
+	// Prepare the additional headers
+	additionalHeaders := make(map[string]string)
```

Comment:
With the changes you're making to the code, perhaps it means that we need to rename this. It does not make sense to call this `additionalHeaders` anymore, because you're actually overwriting some of them. So I would change the name of that variable (and parameter of `writeResponse`).

Reply:
So something like this:
```go
var cacheControlValue string

originalCacheControl := babbageResponse.Header.Get("Cache-Control")

if originalCacheControl != "" {
	cacheControlValue = fmt.Sprintf("%s, max-age=%d", originalCacheControl, maxAge)
} else {
	cacheControlValue = fmt.Sprintf("max-age=%d", maxAge)
}

headersToUpdate := map[string]string{
	"Cache-Control": cacheControlValue,
}

writeResponse(ctx, w, babbageResponse, headersToUpdate)
```

And then, of course, change the name of the `writeResponse` parameter.